radarpipeline/datalib/radar_data.py

```diff
@@ -76,9 +77,12 @@ def get_combined_data_by_variable(
                 user_variables = user_data._get_all_variables()
                 for var in variables:
                     if var in user_variables:
-                        var_data = user_data._get_data_by_key(var)
-                        if var_data is not None:
-                            variable_dict.get(var, []).append(var_data.get_data())
+                        var_data_list = user_data.get_data_by_variable(var)
+                        if var_data_list is not None and len(var_data_list) != 0:
+                            if var not in variable_dict:
+                                variable_dict[var] = []
+                            for var_data in var_data_list:
+                                variable_dict[var].append(var_data.get_data())
```

Comment:
One more change that would be preferable is that if the input is string only, it makes sense to return just one dict rather than list of dicts. 
You can just add a couple of lines before return. 

```python
if len(variable_data_list) == 1:
  return variable_data_list[0]
```

Reply:
Same can be done on `get_data_by_user_id`,

```python
        if len(user_data_list)==1:
            return user_data_list[0]
```